use_cases/refresher.go

```diff
@@ -58,3 +51,83 @@ func (r Refresher) Refresh(ctx context.Context) error {
 
 	return nil
 }
+
+func (r Refresher) generator() <-chan models.Pokemon {
+	out := make(chan models.Pokemon)
+
+	go func() {
+		pokemons, err := r.Read()
+
+		if err != nil {
+			panic(err)
+		}
+
+		for _, pokemon := range pokemons {
+			out <- pokemon
+		}
+
+		defer close(out)
+
+	}()
+
+	return out
+
+}
+
+func (r Refresher) ability(pokemons <-chan models.Pokemon) <-chan models.Pokemon {
+	out := make(chan models.Pokemon)
+
+	go func() {
+		for p := range pokemons {
+			urls := strings.Split(p.FlatAbilityURLs, "|")
+			var abilities []string
+			for _, url := range urls {
+				ability, err := r.FetchAbility(url)
+				if err != nil {
+					panic(err)
+				}
+
+				for _, ee := range ability.EffectEntries {
+					abilities = append(abilities, ee.Effect)
+				}
+			}
+			p.EffectEntries = abilities
+
+			out <- p
+
+		}
+		close(out)
+	}()
+
+	return out
+
+}
+
+func fanIn(inputs ...<-chan models.Pokemon) <-chan models.Pokemon {
+	var wg sync.WaitGroup
+	out := make(chan models.Pokemon)
+
+	wg.Add(len(inputs))
+
+	for _, in := range inputs {
+		go func(ch <-chan models.Pokemon) {
+			for {
```

Comment:
Why iterate in an infinite loop if you use a for over a chan, once it is close the for should break by itself.